cmd/worker/eks.go

```diff
@@ -141,9 +141,6 @@ func registerEKSWorkflows(
 	waitELBsDeletionActivity := eksworkflow.NewWaitELBsDeletionActivity(awsSessionFactory)
 	activity.RegisterWithOptions(waitELBsDeletionActivity.Execute, activity.RegisterOptions{Name: eksworkflow.WaitELBsDeletionActivityName})
 
-	deleteStackActivity := awsworkflow.NewDeleteStackActivity(awsSessionFactory)
-	activity.RegisterWithOptions(deleteStackActivity.Execute, activity.RegisterOptions{Name: awsworkflow.DeleteStackActivityName})
-
```

Comment:
**Question**: how did this go through the previous PR's CI test phase?

The worker's initialization process is expected to panic if two activities are registered under the same name, because the already registered check is enabled by default.

Reply:
```shell
pregnor_69605:Pregnor-BanzaiCloud-MBP152019@188.6.25.232/Users/pregnor/development/src/github.com/banzaicloud/pipeline (refactor/unify-eks-node-pool-creation *$)
> make build && build/worker

# Begin: 2020-11-04T18:07:36.771+01:00

go build  -tags "" -ldflags " -L /usr/local/opt/curl/lib -L /usr/local/opt/llvm/lib  -L /usr/local/opt/qt/lib -L /usr/local/opt/ruby/lib -L /usr/local/opt/tcl-tk/lib -L /usr/local/opt/curl/lib -L /usr/local/opt/llvm/lib  -L /usr/local/opt/qt/lib -L /usr/local/opt/ruby/lib -L /usr/local/opt/tcl-tk/lib -X main.version=refactor/unify-eks-node-pool-creation -X main.commitHash=8e3db19fc -X main.buildDate=2020-11-04T18:07:36+0100 -X main.helmVersion=3.4.0" -o build/ ./cmd/...
INFO[0000] starting application                          application=pipeline.worker arch=amd64 build_date="2020-11-04T18:07:36+0100" commit_hash=8e3db19fc compiler=gc environment=production go_version=go1.15.3 os=darwin version=refactor/unify-eks-node-pool-creation
INFO[0000] Cadence domain already exists                 application=pipeline.worker component=cadence-worker environment=production
INFO[0000] No metrics scope configured for cadence worker. Use NoopScope as default.  application=pipeline.worker component=cadence-worker environment=production
DEBU[0000] assembling helm release deleter...            application=pipeline.worker environment=production
DEBU[0000] assembled helm 3 release deleter              application=pipeline.worker environment=production
WARN[0000] failed to add builtin repository, it might already exist  application=pipeline.worker environment=production repoName=stable
WARN[0000] failed to add builtin repository, it might already exist  application=pipeline.worker environment=production repoName=banzaicloud-stable
WARN[0000] failed to add builtin repository, it might already exist  application=pipeline.worker environment=production repoName=loki
INFO[0000] platform helm environment set up, builtin repos added  application=pipeline.worker environment=production
ERRO[0000] activity type "aws-common-delete-stack" is already registered  application=pipeline.worker environment=production file="registry.go:119" func="(*registry).RegisterActivityWithOptions"

# End: 2020-11-04T18:07:48.971+01:00
# Duration: 12s200ms
# Exit code: 0
```

Duplicated the old entry in eks.go to aws.go.

Reply:
Anyway, let's merge this ASAP, but we need to take preventative steps in the future to avoid similar situation.

Reply:
As for prevention: I've discovered the issue while writing a functional test for integrated services that failed because of this (it starts the worker process and launches cadence workflows)

It should be available as a github action soon.

Reply:
Yeah, I thought of the same - running the binaries is a pretty basic, but preferred safety net if feasible, or alternatively testing the initialization of the binaries can catch some of the similar issues.